app/components/profile.jsx

```diff
@@ -94,13 +85,16 @@ export class Profile extends Component {
 
 }
 
-function mapStateToProps ({ session }) {
-  return { session };
+function mapStateToProps ({ users }, { params: { username } }) {
+  return { user: users[username] };
```

Comment:
since we're no longer using `username`, we just need to change this to `currentUser` instead.

final result:

```js
return { user: users[currentUser] };
```

you should also do this in `tracker.jsx` (line 85)

and thats it! hopefully that all made sense. if it didn't feel free to ask questions. id like for you to understand what the problem was too! now if you try to navigate to another profile page, it should work juuuust fine 👍